src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -874,31 +1160,48 @@ public void AssemblyLoading_NativeDependency()
             });
         }
 
-        [Fact]
-        public void AssemblyLoading_Delete()
+        [Theory]
+        [CombinatorialData]
+        public void AssemblyLoading_DeleteAfterLoad1(bool shadowLoad)
+        {
+            Run(shadowLoad, static (DefaultAnalyzerAssemblyLoader loader, AssemblyLoadTestFixture testFixture) =>
+            {
+                StringBuilder sb = new StringBuilder();
+
+                loader.AddDependencyLocation(testFixture.Delta1.Path);
+                _ = loader.LoadFromPath(testFixture.Delta1.Path);
+
+                if (loader is ShadowCopyAnalyzerAssemblyLoader || !ExecutionConditionUtil.IsWindows)
+                {
+                    File.Delete(testFixture.Delta1.Path);
+                }
+                else
+                {
+                    Assert.Throws<UnauthorizedAccessException>(() => File.Delete(testFixture.Delta1.Path));
+                }
+            });
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void AssemblyLoading_DeleteAfterLoad2(bool shadowLoad)
         {
-            Run(static (DefaultAnalyzerAssemblyLoader loader, AssemblyLoadTestFixture testFixture) =>
+            Run(shadowLoad, static (DefaultAnalyzerAssemblyLoader loader, AssemblyLoadTestFixture testFixture) =>
             {
                 using var temp = new TempRoot();
                 StringBuilder sb = new StringBuilder();
 
                 var tempDir = temp.CreateDirectory();
                 var deltaCopy = tempDir.CreateFile("Delta.dll").CopyContentFrom(testFixture.Delta1.Path);
                 loader.AddDependencyLocation(deltaCopy.Path);
-                Assembly delta = loader.LoadFromPath(deltaCopy.Path);
+                Assembly? delta = loader.LoadFromPath(deltaCopy.Path);
 
-                try
+                if (loader is ShadowCopyAnalyzerAssemblyLoader)
```

Comment:
` || !ExecutionConditionUtil.IsWindows`?

Reply:
BTW, is there any reason to use the shadow copy loader on non-Windows platforms? Is the usage of it conditioned accordingly?

Reply:
I _think_ it can be but need to do more digging. Yes we can delete the files today but is that an accident of implementation or intentional design decision? Also are there other side effects we're not thinking of here? Have to discuss a bit with runtime team. 

Will add it to the list of follow ups in #66532.